include/boost/math/ccmath/signbit.hpp

```diff
@@ -7,19 +7,173 @@
 #define BOOST_MATH_CCMATH_SIGNBIT_HPP
 
 #include <cmath>
+#include <cstdint>
 #include <limits>
 #include <type_traits>
 #include <boost/math/tools/is_constant_evaluated.hpp>
+#include <boost/math/special_functions/detail/fp_traits.hpp>
 #include <boost/math/ccmath/isnan.hpp>
+#include <boost/math/ccmath/abs.hpp>
+
+#if __cpp_lib_bit_cast >= 201806L
```

Comment:
I think we need to include <version> before checking for this macro (and use __has_include to check for <version>).